### ISSUE TITLE
[Bug?]: `yarn set version` doesn't respect current yarn path

### ISSUE THREAD

**Issue by appsforartists** [yarnpkg/berry#5911]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

For compliance reasons, all external code in a repo [must be in `third_party`](https://opensource.google/documentation/reference/releasing/preparing#third-party_components); therefore, this is my yarn path:

```
yarnPath: third_party/.yarn/releases/yarn-3.2.1.cjs
```

when running `yarn set 4.0.1`, the upgrade was mistakenly placed in 

```
.yarn/releases/yarn-4.0.1.cjs
```

### To reproduce

```bash
echo "{}" >> package.json
yarn set version 3.2.1
mkdir third_party
mv ./.yarn ./third_party/.yarn
echo "yarnPath: third_party/.yarn/releases/yarn-3.2.1.cjs" > .yarnrc.yml
yarn set version 4.0.1
```

### Environment

```shell
System:
    OS: macOS 14.0
    CPU: (8) arm64 Apple M1 Pro
  Binaries:
    Node: 20.9.0 - /private/var/folders/p0/kbnclsq57q96dg1kd3699mvc00_vh7/T/xfs-236b58fd/node
    Yarn: 1.22.19 - /private/var/folders/p0/kbnclsq57q96dg1kd3699mvc00_vh7/T/xfs-236b58fd/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    pnpm: 8.10.0 - ~/.nvm/versions/node/v20.9.0/bin/pnpm
```

**Comment by arcanis** [comment]
Indeed, [it's currently hardcoded to `.yarn/releases`](https://github.com/yarnpkg/berry/blob/master/packages/plugin-essentials/sources/commands/set/version.ts#L207). I think it'd make sense to reuse the existing folder if possible (essentially by doing a `path.dirname` on `yarnPath` if it exists, and fallbacking on `.yarn/releases` otherwise).

Are you interested to make a PR?

**Comment by appsforartists** [comment]
I unfortunately don't have time to dig into it right now.  I've been buried in a Vite migration, and I'm at a summit all next week.

**Comment by ramshad-sha** [comment]
> ### Self-service
> * [ ]  I'd be willing to implement a fix
> 
> ### Describe the bug
> For compliance reasons, all external code in a repo [must be in `third_party`](https://opensource.google/documentation/reference/releasing/preparing#third-party_components); therefore, this is my yarn path:
> 
> ```
> yarnPath: third_party/.yarn/releases/yarn-3.2.1.cjs
> ```
> 
> when running `yarn set 4.0.1`, the upgrade was mistakenly placed in
> 
> ```
> .yarn/releases/yarn-4.0.1.cjs
> ```
> 
> ### To reproduce
> ```shell
> echo "{}" >> package.json
> yarn set version 3.2.1
> mkdir third_party
> mv ./.yarn ./third_party/.yarn
> echo "yarnPath: third_party/.yarn/releases/yarn-3.2.1.cjs" > .yarnrc.yml
> yarn set version 4.0.1
> ```
> 
> ### Environment
> ```shell
> System:
>     OS: macOS 14.0
>     CPU: (8) arm64 Apple M1 Pro
>   Binaries:
>     Node: 20.9.0 - /private/var/folders/p0/kbnclsq57q96dg1kd3699mvc00_vh7/T/xfs-236b58fd/node
>     Yarn: 1.22.19 - /private/var/folders/p0/kbnclsq57q96dg1kd3699mvc00_vh7/T/xfs-236b58fd/yarn
>     npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
>     pnpm: 8.10.0 - ~/.nvm/versions/node/v20.9.0/bin/pnpm
> ```
> 
> ### Additional context
> _No response_
